utils/firebase/auth.ts

```diff
@@ -0,0 +1,32 @@
+import { signInWithPopup } from "firebase/auth";
+import { setDoc, doc, getDoc } from "firebase/firestore";
+import { UserDoc } from "models/users";
+import { auth, authProvider, userRef } from "./init";
+
+export const signin = async () => {
+    try {
+        const { user } = await signInWithPopup(auth, authProvider);
+
+        const getDocRef = await getDoc(doc(userRef, user.uid));
+
+        if (getDocRef.exists()) {
+            return;
+        }
+
+        const newAccInfo: UserDoc = {
+            name: user.displayName || "Not Available",
+            email: user.email || "Not Available",
+            id: user.uid,
```

Comment:
Feels necessary to store the `uid` as a field on the document - it's already associated with the document. Can we just use whatever ID field is already on the document when we get it from Firestore?

Reply:
I use the id to reference the users doc in firestore but I could remove it and pass in the uid from their auth credentials if you want. I just did it this way for simplicity since I was referencing values from the previous `UserDoc` data type to update it.

![image](https://user-images.githubusercontent.com/25259737/204341669-385d69e8-194b-4aa1-9c5d-11b93110eb84.png)

Reply:
I get that some ID is necessary, I'm just wondering if you get a document from Firestore and look at the fields, is the UID already there even if we don't add the `id` field to the document? I.e., getting a user doc would return
```
{
  **uid: string**
  name: string
  email: string
  lists: string[]
}
```
and we could just add the native `uid` field to the type without adding an `id` field to the document. Though looking at work code, it seems like we might have to add some custom code to automatically add the `uid` to retrieved Firestore documents so maybe we just leave this as is for now 🤷🏼‍♂️

Reply:
Discussed offline that this will be done in another PR